### ISSUE TITLE
channel.consume() busy-loops eating all CPU

### ISSUE THREAD

**Issue by wichert** [pika/pika#361]
Again using pika 0.9.9, but current git has the same problem. My code is pretty simple:

```python
credentials = pika.PlainCredentials(username, password)
con = pika.BlockingConnection(pika.ConnectionParameters(
        host=server, credentials=credentials))
channel = con.channel()
channel.exchange_declare(exchange=exchange, exchange_type='topic',
        durable=True)
channel.queue_declare(queue=queue, durable=True)
channel.queue_bind(exchange=exchange, queue=queue, 
        routing_key=routing_key)

for method, properties, body in channel.consume(queue):
  ...
```

which is essentially a straight copy from the documentation. When I run this my process starts eating all CPU. An strace reveals why:

```
poll([{fd=3, events=POLLIN|POLLPRI}], 1, 0) = 0 (Timeout)
poll([{fd=3, events=POLLIN|POLLPRI}], 1, 0) = 0 (Timeout)
poll([{fd=3, events=POLLIN|POLLPRI}], 1, 0) = 0 (Timeout)
poll([{fd=3, events=POLLIN|POLLPRI}], 1, 0) = 0 (Timeout)
poll([{fd=3, events=POLLIN|POLLPRI}], 1, 0) = 0 (Timeout)
poll([{fd=3, events=POLLIN|POLLPRI}], 1, 0) = 0 (Timeout)
poll([{fd=3, events=POLLIN|POLLPRI}], 1, 0) = 0 (Timeout)
poll([{fd=3, events=POLLIN|POLLPRI}], 1, 0) = 0 (Timeout)
poll([{fd=3, events=POLLIN|POLLPRI}], 1, 0) = 0 (Timeout)
poll([{fd=3, events=POLLIN|POLLPRI}], 1, 0) = 0 (Timeout)
poll([{fd=3, events=POLLIN|POLLPRI}], 1, 0) = 0 (Timeout)
```

looking at the code this makes sense: select.poll() takes a timeout in milliseconds and the code passing in a timeout value of 0.01 (directly in pika 0.9.9, in a roundabout way in current master by doing 10 / 1000). 

For current master this can be fixed by removing the division by 1000.

**Comment by vitaly-krugl** [comment]
Addressed in the new BlockingConnection implementation.

**Comment by julio-vaz** [comment]
Any update on this matter?

**Comment by vitaly-krugl** [comment]
@julio-vaz, this should be addressed as of pika 0.10.0.

@gmr: let's close this issue. Thx.
